File: patterns/java/src/refactoring_guru/patterns/adapter/example/round/RoundHole.java
package refactoring_guru.patterns.adapter.example.round;

public class RoundHole {
    private double radius;

    public RoundHole(double radius) {
        this.radius = radius;
    }

    public double getRadius() {
        return radius;
    }

    public boolean fits(RoundPeg peg) {
        boolean result;
        result = (this.getRadius() >= peg.getRadius());
        return result;
    }
}

File: patterns/cs/src/RefactoringGuru/Adapter/Example/Round/RoundHole.cs
namespace RefactoringGuru.Adapter.Example.Round
{
    /**
     * EN: RoundHoles are compatible with RoundPegs.
     * 
     * RU: КруглоеОтверстие совместимо с КруглымиКолышками.
     */
    class RoundHole
    {
        private double radius;

        public RoundHole(double radius)
        {
            this.radius = radius;
        }

        public double GetRadius()
        {
            return radius;
        }

        public bool Fits(RoundPeg peg)
        {
            bool result;
            result = this.GetRadius() >= peg.GetRadius();
            return result;
        }
    }
}
